main.cpp

```diff
@@ -12,25 +12,34 @@ int main() {
 
     // Show the network address
     const char *ip = net.get_ip_address();
-    printf("IP address is: %s\n", ip ? ip : "No IP");
+    const char *netmask = net.get_netmask();
+    const char *gateway = net.get_gateway();
+    printf("IP address: %s\n", ip ? ip : "None");
+    printf("Netmask: %s\n", netmask ? netmask : "None");
+    printf("Gateway: %s\n", gateway ? gateway : "None");
 
     // Open a socket on the network interface, and create a TCP connection to mbed.org
     TCPSocket socket;
     socket.open(&net);
-    socket.connect("developer.mbed.org", 80);
+    socket.connect("api.ipify.org", 80);
+    char *buffer = new char[256];
```

Comment:
If we weren't using this RTOS, I'd be quite inclined to use the stack. I probably would have done in mbed OS 3. (And I think it's more predictable+reliable than the heap - the heap can fill up or become fragmented while running, the stack can't.)

But in this RTOS environment, stack usage can be particularly problematic. You need to allocate enough RAM for the peak usage of every thread. 

    Stack RAM = Sum<1..N> [MaxStackUsageForThread(N)]

You need to do everything possible to make each thread's stack usage be as small as possible to try to keep that sum down. It gets out of hand incredibly fast. Ideally, do not create any threads. Increasing the stack usage of any thread increases total RAM requirement.

If you avoid using the RTOS, and use event queue type mechanisms on single thread, then stack RAM would be more like

    Stack RAM = Max<1..N> [MaxStackUsageForEvent(N)]

Which is a /much/ smaller number, and increasing the stack usage of any one event is likely to actually just use already-paid-for stack, unless it becomes the biggest user.

In summary - event loop stacks or the heap are widely-shared resources - by using them for a temporary, you're likely to not increase total system RAM requirement. Thread stacks are less shared, and using them is more likely to increase total system RAM requirement - avoid them.

And to give some sense of scale - RTX thinks a sensible default stack size for a thread is 512 bytes. Not much room for 256-byte buffers on there.